src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +266,47 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Add any pending ICE candidates from the queue for the client ID
+        Queue<IceCandidate> pendingIceCandidatesQueueByClientId = pendingIceCandidatesMap.get(clientId);
```

Comment:
You need to check the hashmap behavior when the entry is missing. Does it throw or does it return null? You should handle the case where the entry doesn't exist.

Reply:
When entry is missing, the value is NULL. So, the case is automatically handled

Reply:
If the value is null (per doc https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#get-java.lang.Object-) you will null deref. Needs an explicit check.

while (pendingIceCandidatesQueueByClientId != NULL && !pendingIceCandidateQueueByClientId.isEMpty())